#########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################

mass        : 2.67 # kg 
gra         : 9.81 
pose_solver : 1     # 0:From ZhepeiWang (drag & less singular) 1:From ZhepeiWang, 2:From rotor-drag    
ctrl_freq_max   : 100.0
use_bodyrate_ctrl: false
max_manual_vel: 1.0
max_angle: 30  # Attitude angle limit in degree. A negative value means no limit.
low_voltage: 13.2 # 4S battery

rc_reverse: # *
    roll: false
    pitch: false
    yaw: false
    throttle: false

auto_takeoff_land:
    enable: true
    enable_auto_arm: true
    no_RC: false
    takeoff_height: 0.5 # m
    takeoff_land_speed: 0.3 # m/s

thrust_model: # The model that maps thrust signal u(0~1) to real thrust force F(Unit:N): F=K1*Voltage^K2*(K3*u^2+(1-K3)*u). 
    print_value: false # display the value of “thr_scale_compensate” or “hover_percentage” during thrust model estimating.
    accurate_thrust_model: false  # This can always enabled if don't require accurate control performance :-)
    # accurate thrust mapping parameters
    K1: 0.7583 # Needs precise calibration!
    K2: 1.6942 # Needs precise calibration!
    K3: 0.6786 # Needs precise calibration! K3 equals THR_MDL_FAC in https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html.
    # approximate thrust mapping parameters
    hover_percentage: 0.35  # Thrust percentage in Stabilize/Arco mode # *

gain: 
    # Cascade PID controller. Recommend to read the code.
    Kp0: 2.0
    Kp1: 2.0 
    Kp2: 2.0
    Kv0: 1.5
    Kv1: 1.5
    Kv2: 1.5
    # ↓↓↓ No use now --
    Kvi0: 0.0
    Kvi1: 0.0
    Kvi2: 0.0
    Kvd0: 0.0
    Kvd1: 0.0
    Kvd2: 0.0
    # ↓↓↓ Only used in rate control mode.
    KAngR: 20.0
    KAngP: 20.0
    KAngY: 20.0

rotor_drag:  
    x: 0.0  # The reduced acceleration on each axis caused by rotor drag. Unit:(m*s^-2)/(m*s^-1).
    y: 0.0  # Same as above
    z: 0.0  # Same as above
    k_thrust_horz: 0.0 # Set to 0 recommended... --

msg_timeout:
    odom: 0.5
    rc:   0.5
    cmd:  0.5
    imu:  0.5
    bat:  0.5
